documentation/src/main/asciidoc/mapstruct-reference-guide.asciidoc

```diff
@@ -293,6 +293,17 @@ Supported values are:
 If a component model is given for a specific mapper via `@Mapper#componentModel()`, the value from the annotation takes precedence.
 |`default`
 
+|`mapstruct.defaultInjectionStrategy`
+| The type of the injection in mapper via parameter `uses`.
+
+Supported values are:
+
+* `field`: dependencies will be injected in fields
+* `constructor`: dependencies will be injected via constructor.
```

Comment:
perhaps state that a constructor will be generated for this. I was thinking last week though: it should also generate an empty constructor for CDI (does it?), otherwise the object cannot be proxied.

Reply:
You're right. componentModel=cdi will generate two constructors: with inject value and default constructor without parameters. I'll add it in documentation.

Reply:
@sjaakd I'm trying to create tests with CDI, but I can't. Maven (I don't know why) doesn't run my simple 'bad" test. But it compiled it (because I had compile errors with bad imports). I have attached picture. Maybe you know the reason? I use: mvn clean install . Tets 
![cdiTest](https://user-images.githubusercontent.com/4740207/64067542-cbbb6d80-cc32-11e9-8514-c48ca0c7ed91.png)

Reply:
It is more tricky than that. I tried to the same with `weld-unit` a while ago. Point is: we use a test runner `@RunWith` to trigger annotation processing and we control which classes we want on the class path (we can extend that with `@WithClasses`. Checkout the `testutil`  package for the details.

So, to use another `@RunWith` will not trigger annotation processing. Hence the implementation classes you need will never be generated. 

To solve this, you probably need to use [junit5](https://junit.org/junit5/docs/current/user-guide/#extensions). I believe that @filiphr has tried that. But that's already a tricky job. We probably need the ` @ExtendWith` to do that and use `Order` to first trigger annotation processing with our own runner and then have weld discover the generated implementations.

Reply:
Ow... I understand. So, about this PR. I think it's better to wait and don't merge it without tests CDI. Should I close?

Reply:
by the way: I gave up on writing a unit test for that one for now. It is too tricky. But if you want a unit test, you could also go for a fixture. Checkout the way `@Rule` is used in `ConstantsTest` [here](https://github.com/mapstruct/mapstruct/blob/master/processor/src/test/java/org/mapstruct/ap/test/source/constants/ConstantsTest.java).

Reply:
> Should I close?

No.. I think your solution is sound. Lets wait for @filiphr's review.